oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/oidc/userinfo/OidcUserRequestUtils.java

```diff
@@ -91,9 +91,10 @@ static OidcUser getUser(OidcUserRequest userRequest, OidcUserInfo userInfo) {
 			authorities.add(new SimpleGrantedAuthority("SCOPE_" + scope));
 		}
 		if (StringUtils.hasText(userNameAttributeName)) {
+			// TODO: Get name from OidcUserAuthority.collectClaims.
```

Comment:
Since this is simply combining the two maps, perhaps you could duplicate that logic here in order to obtain the name

Reply:
Does anything speak against making `OidcUserAuthority.collectClaims` public instead of duplicating the logic in your opinion? I could imagine that by deprecating the `nameAttributeKey` constructors users of Spring Security might also require a snippet such as

```java
Map<String, Object> attributes =  OidcUserAuthority.collectClaims(userRequest.getIdToken(), userInfo);
String name = (String) attributes.get(userNameAttributeName);
new DefaultOidcUser(name, userRequest.getIdToken(), userInfo, authorities);
```

when instantiating `DefaultOidcUser`. With `OidcUserAuthority.collectClaims` being package-private merging the claims would need to be reimplemented for each project.

Reply:
Should no longer be relevant with the introduction of builder classes.